New-comm-layer-step-4

ChristopheLarchier opened the pull request:
Replace xpc methods by generic ones.

ChristopheLarchier committed:
Use generic xpc methods

luc-guyot-infomaniak commented on src/server/comm/guicommserver_mac.mm:
```diff
@@ -267,110 +246,106 @@ - (void)processType:(void (^)(ProcessType))callback {
+    NSString *queryStr = [NSString stringWithCString:query.c_str() encoding:[NSString defaultCStringEncoding]];
+    NSData *queryData = [queryStr dataUsingEncoding:NSUTF8StringEncoding];
+
+    [(GuiLocalEnd *) channel->_privatePtr->localEnd setCurrentRequestId:@(0)];
```
If `_privatePtr` is private, one should use a public accessor to `localEnd`.

ChristopheLarchier replied:
Why?
runSendQuery is a static method of GuiCommChannel so it can access to _privatePtr which is a protected attribute of GuiCommChannel (otherwise it wouldn't compile).

luc-guyot-infomaniak commented on src/server/comm/guicommserver_mac.mm:
```diff
@@ -267,110 +246,106 @@ - (void)processType:(void (^)(ProcessType))callback {
+
+    [(GuiLocalEnd *) channel->_privatePtr->localEnd setCurrentRequestId:@(0)];
+
+    [(GuiLocalEnd *) channel->_privatePtr->localEnd sendQuery:queryData
```
Same remark.

ChristopheLarchier replied:
Why?
runSendQuery is a static method of GuiCommChannel so it can access to _privatePtr which is a protected attribute of GuiCommChannel (otherwise it wouldn't compile).

luc-guyot-infomaniak commented on test/server/comm/testguicommchannel.cpp:
```diff
@@ -180,39 +181,43 @@ void TestGuiCommChannel::testLoginRequestToken() {
-    test(inputParamsStr, channel);
+    test(queryStr, answerStr, channel);
 #else
+    const auto queryStr{Str(R"({ "num": 1,)" // RequestNum::LOGIN_REQUESTTOKEN
```
The fact that we don't need a request identifier on MacOS could be emphasized.

ChristopheLarchier replied:
Comment added.
[7c04e52](https://github.com/Infomaniak/desktop-kDrive/pull/1097/commits/7c04e52a4cf372a329737cc74b5c9b110f578723)

ChristopheLarchier committed:
C/S enum isolation

ChristopheLarchier committed:
Code reorganization

ChristopheLarchier committed:
Address comment

ChristopheLarchier committed:
C/S enum isolation

herve-er committed:
fix(userinfo): fix avatar image conversion to CommBLOB

Added QImageWriter and QBuffer includes for better image processing. Updated UserInfo::toDynamicStruct to save the avatar in PNG format using QBuffer, ensuring proper formatting before conversion to CommBLOB.

ChristopheLarchier committed:
New job added

ChristopheLarchier committed:
Merge branch 'develop' into New-comm-layer-step-4

ChristopheLarchier committed:
Refactoring of GUI job test code

luc-guyot-infomaniak reviewed (commented):
Looks good to me. Some of the code smells (SonarCloud) with the easiest fix could be addressed before merging.

ChristopheLarchier committed:
Address comment

ClementKunz commented on src/libsyncengine/jobs/abstractjob.h:
```diff
@@ -46,6 +46,7 @@ class AbstractJob : public Poco::Runnable {
             _additionalCallback = newCallback;
         }
 
+        void setExitInfo(const ExitInfo &exitInfo) { _exitInfo = exitInfo; }
```
Following https://github.com/Infomaniak/desktop-kDrive/pull/1092, `_exitInfo` should set only using the returned value of  `runJob`. Therefor, no setter should be needed for this attribute.

ChristopheLarchier replied:
Removed.
[c52aee3](https://github.com/Infomaniak/desktop-kDrive/pull/1097/commits/c52aee30b5f56a323d7189ab4fe111563e8e22da)

ChristopheLarchier committed:
Address comment
